internal/api/permissions.go

```diff
@@ -72,6 +90,169 @@ func (r *Router) checkAction(c echo.Context) error {
 	return c.JSON(http.StatusOK, echo.Map{})
 }
 
+type checkPermissionsRequest struct {
+	Actions []checkAction `json:"actions"`
+}
+
+type checkAction struct {
+	ResourceID string `json:"resource_id"`
+	Action     string `json:"action"`
+}
+
+type checkStatus struct {
+	Resource types.Resource
+	Action   string
+	Error    error
+}
```

Comment:
I think I'd break this up into two structs: A `checkRequest` containing the resource/action pair and a `checkResult` containing the `checkRequest` object plus an error.

Reply:
Updated